core/components/molecules/dialog/dialog.js

```diff
@@ -34,75 +35,174 @@ const createButtonForAction = (action, index) => {
+          </DialogBox>
+        </FocusTrap>
+      </Overlay>
+    )
+  }
+}
+
+const DialogBox = styled.div`
   position: relative;
-  overflow: hidden;
-  width: ${props => props.width}px;
-  background: ${colors.base.white};
+  max-height: calc(100vh - (${spacing.xlarge} * 2));
+  display: flex;
+  flex-direction: column;
+  background-color: ${colors.base.white};
   border-radius: 3px;
   box-shadow: 0 5px 15px rgba(0, 0, 0, 0.5);
 `
 
-const DialogTitleBar = styled.div`
-  display: flex;
+const DialogClose = styled.div`
+  position: absolute;
+  top: 0;
+  right: 0;
+  z-index: 1;
+
+  /* Overwirtes the color of the icons */
+  ${Button.Element} {
+    &,
+    i,
+    svg,
+    path {
+      color: ${colors.base.black};
+      fill: ${colors.base.black};
+    }
+  }
+`
+
+const DialogHeader = styled.header`
   position: relative;
-  overflow: hidden;
-  padding: ${spacing.small};
-  padding-bottom: 0;
+  padding: ${spacing.small} ${spacing.large} ${spacing.xsmall} ${spacing.large};
   color: ${colors.text.default};
-  font-weight: ${fonts.weight.medium};
+  word-break: break-word;
+  white-space: pre-wrap;
+  word-wrap: break-word;
   text-align: center;
-  span {
-    flex: 1;
-  }
-  & ${Icon.Element} {
-    flex: none;
-    cursor: pointer;
-  }
+`
+
+const DialogTitle = styled(Heading)`
+  font-weight: ${fonts.weight.medium};
+  font-size: ${fonts.size.default};
+  margin: 0;
 `
 
 const DialogBody = styled.div`
-  padding: ${spacing.medium} ${spacing.large};
+  padding: ${spacing.small} ${spacing.medium} ${spacing.large} ${spacing.medium};
+  flex: 1 1 auto;
+  overflow-y: auto;
+  -webkit-overflow-scrolling: touch;
+  overscroll-behavior: contain;
+  word-break: break-word;
+  word-wrap: break-word;
+
+  /* Clears the margin of the last item of the body */
+  > * {
+    margin-bottom: 0;
+  }
 `
 
-const DialogFooter = styled.div`
+const DialogFooter = styled.footer`
   display: flex;
+  flex: 0 0 auto;
   justify-content: center;
   padding: ${spacing.small};
   border-top: 1px solid ${colors.base.grayLight};
 `
 
 Dialog.Action = DialogAction
-Dialog.Element = DialogElement
-
+Dialog.Element = DialogBox
 Dialog.propTypes = {
+  /** Buttons that will be shown on the dialog's footer */
   actions: PropTypes.arrayOf(
     PropTypes.oneOfType([PropTypes.instanceOf(DialogAction), PropTypes.element])
-  ).isRequired,
-  title: PropTypes.string.isRequired,
-  width: PropTypes.number,
-  onClose: PropTypes.func
+  ),
+  /** Dialog's header title */
+  title: PropTypes.string,
+  /** Dialog's header title heading element */
+  titleElement: PropTypes.oneOf([1, 2, 3, 4]),
+  /** Dialog's container width */
+  width: PropTypes.oneOfType([PropTypes.number, PropTypes.oneOf(Object.keys(overlayContentSizes))]),
+  /** Called when the user clicks on the header's cross or outside the dialog */
+  onClose: PropTypes.func,
+  /** Whether you're presenting a form or a destructive action */
+  role: PropTypes.oneOf(['form', 'destructive'])
```

Comment:
Is there a custom role possible for other dialog cases?

Also, please add examples to explain what this means and does. I couldn't figure what this is

Reply:
The role of the dialog helps to define where the focus should be at the time of presentation or to add additional accesibility tags. You need to set it to `form` when you have a form inside so the focus will go to the first input. In case is `destructive` the focus will be on the close button (as default) but the `aria-describedby` attribute will be set to `dialog-description`.

On other cases, no role is required. You can use the dialog as normal.

Reply:
Stories for both roles and default are present.